### ISSUE TITLE
Pausing and Unpausing the Game in Subspace stops the Subspace Sound

### ISSUE THREAD

**Issue by AndrewofDoom** [scp-fs2open/fs2open.github.com#1433]
Recreate by playing a mission that takes place in subspace and pause and unpause the game.

**Comment by asarium** [comment]
I can't reproduce this. I made a simple mission which only has the subspace flag enabled and nothing else. What mission are you using to reproduce it?

**Comment by MageKing17** [comment]
After further examination, the problem is with pausing via the escape menu, rather than pausing normally; a regular pause will involve a pair of state changes which will stop and then restart the subspace ambient sounds. If you hit escape, however, there's no state change, and looping sounds are stopped and never restarted.

Not sure how to fix this properly; changing the way the escape menu works sounds rife for regressions.

**Comment by asarium** [comment]
Even if I use the escape menu the subspace sound still comes back after I close the menu again so I still can't reproduce this issue.

**Comment by FSCyborg** [comment]
You were probably using debug builds @asarium, void game_simulation_frame() has

```
#ifndef NDEBUG
		if ( Game_subspace_effect ) {
			game_start_subspace_ambient_sound();
		}
#endif
```

It works in debug, besides the (I would assume negligible) performance hit is there a big downside to just enabling it this block on all builds?

**Comment by asarium** [comment]
I guess that is some Volition legacy code sincr only enabling that in debug makes absolutely no sense.

Feel free to submit a PR that removes the ifdefs.

**Comment by FSCyborg** [comment]
All right

**Comment by Goober5000** [comment]
It makes perfect sense. It is for when subspace is enabled or disabled through the debug console, which is only active in debug builds.

The proper way to fix this is by adding the appropriate hook to the Escape key.

**Comment by FSCyborg** [comment]
I did a quick check and the target lock sound starts back up after exiting the menu, so this is probably the only sound not restarting correctly.  

How's this? I only had to change things in `case 2:`

```
//Cyborg17 - Linking this function to restart subspace sound if needed. This is to avoid having to
//check for subspace every frame.
extern void game_start_subspace_ambient_sound();

/**
 * Handler for when player hits 'ESC' during the game
 */
void game_do_end_mission_popup()
{
	int	pf_flags, choice;

	// do the multiplayer version of this
	if(Game_mode & GM_MULTIPLAYER){
		multi_quit_game(PROMPT_ALL);
	} else {
		// single player version....
		// do housekeeping things.
		game_stop_time();
		game_stop_looped_sounds();
		snd_stop_all();

		pf_flags = PF_BODY_BIG | PF_USE_AFFIRMATIVE_ICON | PF_USE_NEGATIVE_ICON;
		choice = popup(pf_flags, 3, POPUP_NO, XSTR( "&Yes, Quit", 28), XSTR( "Yes, &Restart", 29), XSTR( "Do you really want to end the mission?", 30));

		switch (choice) {
		case 1:
			gameseq_post_event(GS_EVENT_END_GAME);
			break;

		case 2:
			if (Game_subspace_effect) {
				game_start_subspace_ambient_sound();
			}
			gameseq_post_event(GS_EVENT_ENTER_GAME);
			break;

		default:
			break;  // do nothing
		}

		game_start_time();
		game_flush();
	}
}
```

**Comment by Goober5000** [comment]
That looks like a much better candidate for the root cause.  However I think it would be better to put those changes beneath `game_start_time` to mirror the top of the function.

On that same subject, is there a `game_restart_looping_sounds` function?  Do any of the other sounds stopped from `game_stop_looping_sounds` need to be restarted?

**Comment by FSCyborg** [comment]
I looked for one, but couldn't find one.

**Comment by Goober5000** [comment]
The `gameseq_post_event` function tells the game engine "Switch to this game state on the next frame".  The switch doesn't happen immediately; the function exits as normal.  And all the processing and post-processing of that game frame cleans up as normal too.

In short: yes.

**Comment by Goober5000** [comment]
Do any of the other sounds stopped from `game_stop_looping_sounds` need to be restarted, though?  This is an important question, because if there is one (the subspace sound), there may be others.

**Comment by FSCyborg** [comment]
Like I said, I tested for missile lock sound.  I'll do some quick tests to see if the other loops are rest.  @MageKing17 @asarium Do you see any issues with this approach?

**Comment by FSCyborg** [comment]
Here are the sounds we need to check.

	hud_stop_looped_locking_sounds();       _Works Correctly_
	hud_stop_looped_engine_sounds();        _Works Correctly_
	afterburner_stop_sounds();           _Afterburner is stopped when escape key is pressed, so its sounds do not need to be played_
	player_stop_looped_sounds();         _Need to figure out what these are.._...
	obj_snd_stop_all();		// stop all object-linked persistant sounds _Need to figure out what these are.._...
	game_stop_subspace_ambient_sound();     _To be fixed_
	snd_stop(Radar_static_looping);  _?????_
	Radar_static_looping = sound_handle::invalid();  _??????_
	snd_stop(Target_static_looping);   _??????_
	shipfx_stop_engine_wash_sound();    _To be tested_
	Target_static_looping = sound_handle::invalid();  _??????_

**Comment by FSCyborg** [comment]
I'm just going to submit the PR.

**Comment by Goober5000** [comment]
Fixed via #2078
